### ISSUE TITLE
Target JVM version does not align in Gradle for Kotlin

### ISSUE THREAD

**Issue by sohrab-** [spring-io/initializr#1084]
Below configuration:

![image](https://user-images.githubusercontent.com/4444588/80768859-e44ba100-8b8e-11ea-9f84-2b4a0f522898.png)

Generates the following Gradle script:

![image](https://user-images.githubusercontent.com/4444588/80768841-dac23900-8b8e-11ea-92cb-8786480bf5da.png)

There is a mismatch between line 12:

```kotlin
java.sourceCompatibility = JavaVersion.VERSION_11
```

and line 34:

```kotlin
jvmTarget = "1.8"
```

Probably a good one for a first contribution. I myself am not familiar with the code yet so if I get some bandwidth, we'll pull request.

**Comment by LifeIsStrange** [comment]
Also I guess that implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8") should match the targeted minimum jdk

**Comment by sohrab-** [comment]
No, I don't think so. `stdlib-jdk8` is correct for targeting 11. There aren't corresponding ones for each version after 8.

Although the preferred format for those Kotlin dependencies is now:

```kotlin
implementation(kotlin("reflect"))
implementation(kotlin("stdlib-jdk8"))
```

**Comment by LifeIsStrange** [comment]
@sohrab- I didn't new this syntax sugar thx!
Kotlin stable (since 1.3.70) support jvm bytecode target up to openjdk 13
https://youtrack.jetbrains.com/issue/KT-34119
And 1.4 bring openjdk 14  bytecode support.

But you are right kotlin does not currently provide stdlib jdk beyond jdk8 which is a shame and I can't even find an issue on youtrack.jetbrains.com
